samples/helmexample/README.md

```diff
@@ -0,0 +1,138 @@
+
+To set the right subscription for the service, run: 
+```
+az account set --subscription <subscription-id>
+``` 
+To give credentials for the resource group to the kubernetes service, run:
+```
+az aks get-credentials --resource-group <resource-group-name> --name <kubernetes-service-name>
+``` 
+
+## Push an image to the container registry
+The following steps illustrates how to push a webservice image in order to be deployed in AKS using ACR (Azure Container Registry).
+1. Build a `published` image using for instancee the following [Dockerfile](https://docs.microsoft.com/en-us/dotnet/core/docker/build-container?tabs=windows#create-the-dockerfile)
+    ```sh
+    docker build -t myapp:v1 .
+    ```
+1. Login into ACR using the username and password credentials that are needed in order to push. See Access Keys section of the ACR Portal.
+    ```sh
+    docker login <acrloginserver>.azurecr.io
+    ```
+1. Tag the image following ACR tagging scheme
+    ```sh
+    docker tag myapp <acrloginserver>.azurecr.io/myapp:v1
+    ```
+1. Push to ACR
+    ```sh
+    docker push <acrloginserver>.azurecr.io/myapp:v1
+    ```
+
+After these steps, using Azure's Portal ACR resource an image should be available under repositories following the naming convention mentioned above.
+
+### Give cluster access to ACR
+To attach the ACR to the cluster so that the image can be accessed, run 
+```
+az aks update -n <cluster-name> -g <resource-group-name> --attach-acr <acr-name>
+```
+
+## Create a new Helm chart (optional)
+The easiest way to create a helm chart is to use a VSCode dev container and when you set it up, ensure to set it up with the latest `debian` OS. For add-on packages, make sure to select the `azure CLI` and `kubernetes-helm` options.
+
+Once you've got the dev container open, run `helm` to ensure you have the helm CLI running properly. Then you can create a new helm chart by running
+```
+helm create <chart-name>
+```
+
+## Setting up the Helm chart
+Once you've got your helm chart open (whether from scratch or existing), the main files you will likely be working with are `Chart.yaml`, `values.yaml`, and `deployment.yaml`.
+
+### Chart.yaml
+In `Chart.yaml`, we won't need to make any changes but make note of the chart `name`, as you will need to reference it in commands later on.
+
+### Values.yaml
+In `values.yaml`, we need to change a couple fields. In the `image` section, you will need to set
+* The `repository` field to be `<acr-login-server>/<image-repository>`
+* The `pullPolicy` field to be `IfNotPresent` (pulls a new image if not present) or `Always` (pulls a new image every time).
+* The `tag` field if you need a particuar tag version
+
+Set the `nameOverride` and `fullNameOverride` fields to make it easier to reference your helm chart, and ensure they are not the same.
+
+In the `serviceAccount` section, ensure that
+* The `name` field is set to the name of the helm chart from `Chart.yaml`.
+
+### Deployment.yaml
```

Comment:
Should we make this variables from Values.Yaml?  Generally not considered to be a good thing to need to modify stuff in "templates" as a standard process.

Reply:
Updates this and the readme accordingly, thx